### PR TITLE
support setting all HuggingFace TrainingArguments

### PR DESCRIPTION
**Pull request by samos123**
Example Model used for testing:
```
apiVersion: substratus.ai/v1
kind: Model
metadata:
  name: falcon-7b-instruct-k8s
spec:
  image:
    name: substratusai/model-trainer-huggingface:pr-16
  baseModel:
    name: falcon-7b-instruct
  trainingDataset:
    name: k8s-instructions
  params:
    # num_train_epochs: 1
    max_steps: 5
  resources:
    gpu:
      count: 4
      type: nvidia-l4
```

### CODE REVIEW

File: model-trainer-huggingface/src/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+def test_parse_training_args():
+    params = {"num_train_epochs": "1"}
+    assert parse_training_args(params).num_train_epochs == 1.0
+
+    params = {"num_train_epochs": "1", "max_steps": "5"}
+    args = parse_training_args(params)
+    assert args.num_train_epochs == 1.0
+    assert args.max_steps == 5
+    assert args.output_dir == "/content/model/checkpoints"
```
Comment (brandonjbjelland) [inline model-trainer-huggingface/src/test_utils.py:11]: 👏

File: model-trainer-huggingface/src/utils.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        optim="paged_adamw_32bit",
+    )
+
+    training_parameters = inspect.signature(TrainingArguments.__init__).parameters
+    for k, v in training_parameters.items():
+        if k in params:
+            val = params.get(k)
+            if v.annotation == bool:
+                # https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python
+                val = json.loads(str(val).lower())
```
Comment (brandonjbjelland) [inline model-trainer-huggingface/src/utils.py:26]: This works for `"true"`, `"True"`, `"false"`, and `"False"`. Comparing with the former `strtobool` implementation, this is a bit less coverage but probably okay. For the applicable params you're seeing on models, do you imagine users would reach for truthy values of `"y"`, `"yes"`, `"1"`, `"on"`, `"t"`?

**Review by brandonjbjelland — APPROVED** [review]
Excellent! 🥝

File: model-trainer-huggingface/src/utils.py (hunk 2 above)
Comment (samos123) [reply model-trainer-huggingface/src/utils.py:26]: I actually think we should only support what TrainingArguments support which is True and False only. The lower version is something we happen to get due to json. I want it to throw an exception when someone doesn't use true or false, wich is what would happen know I think. Let me write some test case for these scenarios

Comment (brandonjbjelland) [reply model-trainer-huggingface/src/utils.py:26]: Perfectly good by me as long as we're intentional with what we're supporting.

Comment (samos123) [reply model-trainer-huggingface/src/utils.py]: I fixed this and now it throws a helpful error with the specific argument and current value that's not in the right format: https://github.com/substratusai/images/pull/16/files#diff-d1f6d276949b3fed15d1c35f294b7136918094827d2d12c54c438225a4e1dfc3R28

Great feedback to improve UX! Glad you brought this up.

Comment (brandonjbjelland) [reply model-trainer-huggingface/src/utils.py]: perfect

**Review by brandonjbjelland — APPROVED** [review]

**Pull request merged**